### PR TITLE
EVG-3193: make host allocator configurable

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: util/slice.go [hunk 1]
```diff
@@ -2,6 +2,10 @@ package util
 
 // StringSliceContains determines if a string is in a slice
 func StringSliceContains(slice []string, item string) bool {
+	if slice == nil || len(slice) == 0 {
```
Comment (richardsamuels) [inline util/slice.go]: You only need the len == 0

File: service/templates/admin.html [hunk 2]
```diff
@@ -365,6 +365,10 @@ <h2 class="modal-title">[[modalTitle]]</h2>
                     <label>Task finder</label>
                     <input type="text" ng-model="Settings.scheduler.task_finder">
                   </md-input-container>
+                  <md-input-container class="control" style="width:45%; margin-left:50px;">
```
Comment (john-m-liu) [inline service/templates/admin.html]: leave off the margin-left:50px
Those are used for controls on the right side to space things out

File: rest/model/admin.go [hunk 3]
```diff
@@ -944,13 +944,15 @@ func (a *APIRepoTrackerConfig) ToService() (interface{}, error) {
...
-	TaskFinder APIString `json:"task_finder"`
+	TaskFinder    APIString `json:"task_finder"`
+	HostAllocator APIString `json:"host_allocator"`
 }
 
 func (a *APISchedulerConfig) BuildFromService(h interface{}) error {
 	switch v := h.(type) {
 	case evergreen.SchedulerConfig:
 		a.TaskFinder = ToAPIString(v.TaskFinder)
+		a.HostAllocator = ToAPIString(v.HostAllocator)
```
Comment (john-m-liu) [inline rest/model/admin.go]: Need to add this to ToService as well

File: units/scheduler.go [hunk 4]
```diff
@@ -40,20 +43,28 @@ func makeDistroSchedulerJob() *distroSchedulerJob {
...
 }
 
 func (j *distroSchedulerJob) Run(ctx context.Context) {
 	defer j.MarkComplete()
 
+	if j.env == nil {
+		j.env = evergreen.GetEnvironment()
+	}
+
+	settings := j.env.Settings()
```
Comment (john-m-liu) [inline units/scheduler.go]: Should this look up the settings from the db instead? That way you can change the allocator on the fly without having to restart the service

File: config_scheduler.go [hunk 5]
```diff
@@ -10,7 +11,8 @@ var taskFinderKey = "task_finder"
 
 // SchedulerConfig holds relevant settings for the scheduler process.
 type SchedulerConfig struct {
-	TaskFinder string `bson:"task_finder" json:"task_finder" yaml:"task_finder"`
+	TaskFinder    string `bson:"task_finder" json:"task_finder" yaml:"task_finder"`
+	HostAllocator string `bson:"host_allocator" json:"host_allocator" yaml:"host_allocator"`
```
[the remark this answers is not in the document]
Comment (richardsamuels) [reply config_scheduler.go]: You can just  pass the actual struct into set, instead of individually setting each field

**Pull request merged**